libpod/image/docker_registry_options.go

```diff
@@ -0,0 +1,35 @@
+package image
+
+import "github.com/containers/image/types"
+
+// DockerRegistryOptions encapsulates settings that affect how we connect or
+// authenticate to a remote registry.
+type DockerRegistryOptions struct {
+	// DockerRegistryCreds is the user name and password to supply in case
+	// we need to pull an image from a registry, and it requires us to
+	// authenticate.
+	DockerRegistryCreds *types.DockerAuthConfig
+	// DockerCertPath is the location of a directory containing CA
+	// certificates which will be used to verify the registry's certificate
+	// (all files with names ending in ".crt"), and possibly client
+	// certificates and private keys (pairs of files with the same name,
+	// except for ".cert" and ".key" suffixes).
+	DockerCertPath string
+	// DockerInsecureSkipTLSVerify turns off verification of TLS
+	// certificates and allows connecting to registries without encryption.
+	DockerInsecureSkipTLSVerify bool
+}
+
+// GetSystemContext constructs a new system context from the given signaturePolicy path and the
+// values in the DockerRegistryOptions
+func (o DockerRegistryOptions) GetSystemContext(signaturePolicyPath, authFile string, forceCompress bool) *types.SystemContext {
```

Comment:
Is this duplicated below?

Reply:
I think there is a private duplicate of this, minus the DockerRegistryOptions part, in utils.go

Reply:
Can the two be unified? Maybe DockerRegistryOptions an optional parameter?

Reply:
It is like this in the current code base.  To minimize risk on breakage, I would like to let this remain for the time being.

Reply:
I'm really iffy on having two identical functions here, even temporarily. Can we at least move them next to each other and add some TODO comments to rewrite them into one function?

Reply:
not unified but co located now